pkg/controllers/serviceca/service_ca_controller.go

```diff
@@ -0,0 +1,223 @@
+package serviceca
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"OAuthServiceDegraded",
+	"SystemServiceCAConfigDegraded",
+	"AuthServiceCAProgressing",
+)
+
+type serviceCAController struct {
+	services       corev1client.ServicesGetter
+	secrets        corev1client.SecretsGetter
+	configMaps     corev1client.ConfigMapsGetter
+	operatorClient v1helpers.OperatorClient
+}
+
+func NewServiceCAController(kubeInformersForTargetNamespace informers.SharedInformerFactory, configInformer configinformers.SharedInformerFactory,
+	secrets corev1client.SecretsGetter, configMaps corev1client.ConfigMapsGetter, services corev1client.ServicesGetter, operatorClient v1helpers.OperatorClient, recorder events.Recorder) factory.Controller {
+	c := &serviceCAController{
+		services:       services,
+		secrets:        secrets,
+		configMaps:     configMaps,
+		operatorClient: operatorClient,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Informer(),
+		configInformer.Config().V1().Authentications().Informer(),
+		configInformer.Config().V1().Ingresses().Informer(),
+	).ResyncEvery(30*time.Second).WithSync(c.sync).ToController("ServiceCAController", recorder.WithComponentSuffix("service-ca-controller"))
+}
+
+func (c *serviceCAController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	foundConditions := []operatorv1.OperatorCondition{}
+
+	// make sure we create the service before we start asking about service certs
+	_, _, err := resourceapply.ApplyService(c.services, syncCtx.Recorder(), getOAuthService())
```

Comment:
this is a constant, you could move it to bindata now that we have them

Reply:
can this be a follow up? i want to minimize the diff.

Reply:
nvmd. I already moved this to bindata here